File: WgWall.Data/Model/TaskTemplate.cs
using System;
using WgWall.Data.Model.Base;

namespace WgWall.Data.Model
{
    public class TaskTemplate : BaseEntity
    {
        public string Name { get; set; }
        public int? IntervalInDays { get; set; }
        public DateTime? LastExecutionAt { get; set; }
        public bool Hidden { get; set; }

        public static TaskTemplate Create(string name, int? intervalInDays, FrontendUser createdBy)
        {
            var element = new TaskTemplate { Name = name, IntervalInDays = intervalInDays};
            RegisterCreation(element, createdBy);
            return element;
        }
    }
}

File: WgWall/ClientApp/src/app/models/task-template.ts
import { Base } from "./base/base";
import { getLocaleDateTimeFormat } from "@angular/common";

export class TaskTemplate extends Base {
  name: string;
  intervalInDays: number | null;
  lastExecutionAt: string | null;
  reward: number;

  expectedRelativeCompletion: number;
}
